### PR TITLE
feat(APIGuildForumChannel): update and add missing features

### PR DESCRIPTION
**Pull request by Mateo-tem**
**Please describe the changes this PR makes and why it should be merged:**

This would be the second and _final_ part of #398, adding support for forum channels.
> **Note:** Maybe you can see that in the first commit are different descriptions on the fields than in the upstream PR. I've already made a draft of this in the previous days, I just replaced all of those, and it's synced now!

**If applicable, please reference Discord API Docs PRs or commits that influenced this PR:**
- https://github.com/discord/discord-api-docs/pull/5391
- https://github.com/discord/discord-api-docs/pull/5453
- https://github.com/discord/discord-api-docs/pull/5458
- https://github.com/discord/discord-api-docs/pull/5475
- https://github.com/discord/discord-api-docs/pull/5480
- https://github.com/discord/discord-api-docs/pull/5474

### CODE REVIEW

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: deno/rest/v10/channel.ts [hunk 1]
```diff
@@ -141,6 +143,25 @@ export type RESTPatchAPIChannelJSONBody = AddUndefinedToPossiblyUndefinedPropert
 	 * Channel types: privateThread
 	 */
 	invitable?: boolean;
+	/**
+	 * The set of tags that can be used in a forum channel
+	 *
+	 * Channel types: forum
+	 */
+	available_tags?: APIGuildForumTag[];
```
[the remark this answers is not in the document]
Comment (Mateo-tem) [reply deno/rest/v10/channel.ts:151]: What do you mean?

File: deno/rest/common.ts [hunk 2]
```diff
@@ -246,7 +251,10 @@ export enum RESTJSONErrorCodes {
 	MessageWasBlockedByAutomaticModeration = 200000,
 	TitleWasBlockedByAutomaticModeration,
 
-	WebhooksCanOnlyCreateThreadsInForumChannels = 220003,
+	WebhooksPostedToForumChannelsMustHaveAThreadNameOrThreadId = 220001,
+	WebhooksPostedToForumChannelsCannotHaveBothAThreadNameOrThreadId,
```
Comment (vvito7) [inline deno/rest/common.ts:255]:
```suggestion
	WebhooksPostedToForumChannelsCannotHaveBothAThreadNameAndThreadId,
```

File: deno/rest/v10/channel.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (vladfrangu) [reply deno/rest/v10/channel.ts:151]: Well thats the rest route. I was asking is available_tags requires the id property of GuildForumTag or not

[the remark this answers is not in the document]
Comment (Mateo-tem) [reply deno/rest/v10/channel.ts:151]: In the docs says that `available_tags` in the rest route are just "an array of tag objects", so I think yes, it needs the `id` property (correct me if I'm wrong).

[the remark this answers is not in the document]
Comment (vladfrangu) [reply deno/rest/v10/channel.ts:151]: Well you can't have an ID if you're just creating the tags 😕 
Might have to ask in the docs repo

[the remark this answers is not in the document]
Comment (vladfrangu) [reply deno/rest/v10/channel.ts:151]: Just asked, this is fine as is!

**Comment by vladfrangu** [comment]
Per discussing with devs:

- In RESTPostAPIGuildChannelsJSONBody or w/e:
  - `default_thread_rate_limit ` can also be set (once https://github.com/discord/discord-api-docs/issues/5400 is resolved)
  - reaction and tags can also be set on create
- In RESTPatchAPIGuildChannelJSONBody:
  - Please add a documentation warning for the tags properties that the id is *optional* when creating new tags (so they can set it to "0")

**Comment by Mateo-tem** [comment]
> * In RESTPatchAPIGuildChannelJSONBody:
>   
>   * Please add a documentation warning for the tags properties that the id is _optional_ when creating new tags (so they can set it to "0")

I've tried making a `PATCH` request to edit a forum channel and only the `name` property is required for new tags. Then, the id will be generated automatically and will be returned in the `available_tags` property along with the tag objects in the payload (the new forum channel).

**Comment by Mateo-tem** [comment]
> * In RESTPatchAPIGuildChannelJSONBody:
>   
>   * Please add a documentation warning for the tags properties that the id is _optional_ when creating new tags (so they can set it to "0")

- https://github.com/discord/discord-api-docs/pull/5458

**Comment by almeidx** [comment]
Can you add this as well?
- discord/discord-api-docs#5475

**Comment by almeidx** [comment]
Needs some updates to reflect the changes made in
- discord/discord-api-docs#5475

**Comment by almeidx** [comment]
And again!
- https://github.com/discord/discord-api-docs/pull/5480

Maybe just revert the default_sort_order changes so this PR doesn't get blocked for the rest of eternity?

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**